components/user-mgt/org.wso2.carbon.user.mgt/src/test/java/org/wso2/carbon/user/mgt/UserRealmProxyTest.java

```diff
@@ -0,0 +1,277 @@
+        Mockito.when(realm.getUserStoreManager()).thenReturn(userStoreManager);
+        userRealmProxy.changePassword("admin2", "newPassword");
+        Assert.assertTrue(true);
+    }
+
+    @Test
+    public void testDeleteUser() throws Exception {
+
+        Registry registry = mock(Registry.class);
+        Mockito.when(realm.getRealmConfiguration()).thenReturn(this.getSampleRelaimConfiguration());
+        Mockito.when(realm.getAuthorizationManager()).thenReturn(authorizationManager);
+        Mockito.when(realm.getUserStoreManager()).thenReturn(userStoreManager);
+        Mockito.when(authorizationManager.
+                isUserAuthorized(anyString(), eq(PERMISSION), eq(UserMgtConstants.EXECUTE_ACTION))).thenReturn(true);
+        userRealmProxy.deleteUser("testUser", registry);
+        verify(userStoreManager).deleteUser(anyString());
+    }
+
+    @Test(dataProvider = "userListCount")
+    public void testGetRolesOfUser(int listCount, int assertCount) throws Exception {
+        Mockito.when(realm.getUserStoreManager()).thenReturn(userStoreManagerWithAb);
+        Mockito.when(((AbstractUserStoreManager) userStoreManagerWithAb).getRoleListOfUser("admin2")).thenReturn(new String[]{"PRIMARY/role1", "PRIMARY/role2"});
+        Object cc = userStoreManagerWithAb;
+        HybridRoleManager hybridRoleManager = mock(HybridRoleManager.class);
+        Field f1 = cc.getClass().getSuperclass().getDeclaredField("hybridRoleManager");
+        f1.setAccessible(true);
+        f1.set(cc, hybridRoleManager);
+        String[] test = {"PRIMARY/role3", "PRIMARY/role4"};
+        String[] test2 = {"role3", "role4"};
+        Mockito.when(((AbstractUserStoreManager) userStoreManagerWithAb).getRoleNames("role", listCount,
+                true, true, true)).thenReturn(null);
+        Mockito.when(hybridRoleManager.getHybridRoles("role")).thenReturn(test);
+        FlaggedName[] flaggedNames = userRealmProxy.getRolesOfUser("admin2", "role", listCount);
+        Assert.assertEquals(flaggedNames.length, assertCount);
+    }
+
+    @Test
+    public void testUpdateUsersOfRole() throws Exception {
+        Mockito.when(realm.getRealmConfiguration()).thenReturn(this.getSampleRelaimConfiguration());
+        Mockito.when(realm.getAuthorizationManager()).thenReturn(authorizationManager);
+        Mockito.when(realm.getUserStoreManager()).thenReturn(userStoreManager);
+        Mockito.when(authorizationManager.
+                isRoleAuthorized(anyString(), eq(PERMISSION), eq(UserMgtConstants.EXECUTE_ACTION))).thenReturn(true);
+        userRealmProxy.updateRoleName("testRole", "testNewRole");
+        verify(userStoreManager).updateRoleName("testRole", "testNewRole");
+    }
+
+    private static void startTenantFlow(String tenantDomain) {
+        String carbonHome = Paths.get(System.getProperty("user.dir"), "target").toString();
+        System.setProperty("carbon.home", carbonHome);
```

Comment:
Better to use `CarbonBaseConstants.CARBON_HOME` :)

Reply:
Cant use unseen wording for Asserts So this is OK for test code